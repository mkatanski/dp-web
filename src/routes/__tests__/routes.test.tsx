/* eslint-disable react/display-name */
import React, { Suspense } from "react";
import TestRenderer from "react-test-renderer";
import { wait, cleanup } from "@testing-library/react";

import Routes from "../routes";

jest.mock(
  "pages/errors/NotFoundPage",
  () => ({ children, ...rest }: React.PropsWithChildren<{}>) => (
    <div id="NotFoundPage" {...rest}>
      {children}
    </div>
  )
);

jest.mock(
  "pages/public/HomePage",
  () => ({ children, ...rest }: React.PropsWithChildren<{}>) => (
    <div id="HomePage" {...rest}>
      {children}
    </div>
  )
);

jest.mock(
  "layouts/DefaultLayout",
  () => ({ children, ...rest }: React.PropsWithChildren<{}>) => (
    <div id="DefaultLayout" {...rest}>
      {children}
    </div>
  )
);

const waitForLazyImports = async (importsNumber: number) => {
  for (let i = 0; i < importsNumber; i++) {
    await wait();
  }
};

describe("routes", () => {
  beforeEach(cleanup);

  it("should render routes schema", async () => {
    const tree = TestRenderer.create(
      <Suspense fallback={"loading"}>
        <Routes />
      </Suspense>
    );

    await waitForLazyImports(2);

    expect(tree).toMatchSnapshot();
  });
});
