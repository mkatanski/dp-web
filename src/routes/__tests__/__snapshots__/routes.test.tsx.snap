// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes should render routes schema 1`] = `
<div
  id="ReactRouter.Switch"
>
  <div
    id="DefaultLayout"
    path="/*"
  >
    <div
      exact={true}
      id="ReactRouter.Route"
      path="/"
    >
      <div
        id="ReactRouter.Redirect"
        to="/deployments"
      />
    </div>
    <div
      exact={true}
      id="ReactRouter.Route"
      path={
        Array [
          "/deployments",
        ]
      }
    >
      <div
        id="HomePage"
      />
    </div>
    <div
      exact={true}
      id="ReactRouter.Route"
      path={
        Array [
          "/templates",
        ]
      }
    >
      <div
        id="TemplatesPage"
      />
    </div>
    <div
      id="ReactRouter.Route"
    >
      <div
        id="NotFoundPage"
      />
    </div>
  </div>
  <div
    id="ReactRouter.Route"
  >
    <div
      id="NotFoundPage"
    />
  </div>
</div>
`;
